### PR TITLE
refactoring according to golangci-lint checks

### PR DESCRIPTION
**Pull request by FriedrichWilken**
@k15r noticed that `golangci-lint checks` shows following minor issues with the eventing `eventing-controllers` code base.

```console
#################################################################################################
# Thu Nov 11 17:08:50 CET 2021
# Run golangci-lint checks
#################################################################################################

Checks: deadcode errcheck gosimple govet ineffassign staticcheck structcheck typecheck unused varcheck golint gofmt misspell gochecknoinits unparam scopelint gosec
WARN [runner] The linter 'scopelint' is deprecated (since v1.39.0) due to: The repository of the linter has been deprecated by the owner.  Replaced by exportloopref.
testing/matchers.go:169: File is not `gofmt`-ed with `-s` (gofmt)
			return s.Status.CleanEventTypes },
testing/test_helpers.go:464:6: func ToUnstructuredApiRule should be ToUnstructuredAPIRule (golint)
func ToUnstructuredApiRule(obj interface{}) (*unstructured.Unstructured, error) {
     ^
reconciler/subscription/nats/reconciler_test.go:415:1: context.Context should be the first parameter of a function (golint)
func ensureSubscriptionCreated(subscription *eventingv1alpha1.Subscription, ctx context.Context) {
^
reconciler/subscription/nats/reconciler_test.go:435:1: context.Context should be the first parameter of a function (golint)
func ensureSubscriptionUpdated(subscription *eventingv1alpha1.Subscription, ctx context.Context) {
^
reconciler/subscription/nats/reconciler_test.go:470:1: context.Context should be the first parameter of a function (golint)
func getSubscription(subscription *eventingv1alpha1.Subscription, ctx context.Context) AsyncAssertion {
^
reconciler/subscription/nats/reconciler_test.go:511:2: var `testId` should be `testID` (golint)
	testId            int
	^
reconciler/subscription/nats/reconciler_test.go:512:2: var `natsUrl` should be `natsURL` (golint)
	natsUrl           string
	^
reconciler/subscription/nats/reconciler_test.go:556:2: var `natsUrl` should be `natsURL` (golint)
	natsUrl := natsServer.ClientURL()
	^
reconciler/subscription/nats/reconciler_test.go:614:20: func parameter `natsUrl` should be `natsURL` (golint)
func connectToNats(natsUrl string) (*nats.Conn, error) {
                   ^
make: [lint] Error 1 (ignored)
```

This PR tries to fix these issues.

### CODE REVIEW

**Comment by pxsalehi** [comment]
You should bump the image to the PR image.

**Review by k15r — DISMISSED** [review]

**Comment by k15r** [comment]
related issue: 12468

**Comment by k15r** [comment]
https://github.com/kyma-project/kyma/pull/12599/commits/96fd691c3c0323f0a19753966e261b67c2bf533f requires https://github.com/kyma-project/kyma/pull/12598 to be merged to be fully working

File: resources/eventing/values.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ global:
       version: PR-11920
     eventing_controller:
       name: eventing-controller
-      version: 571bf38e
+      version: PR-12553
```
Comment (k15r) [inline resources/eventing/values.yaml:11]:
```suggestion
      version: PR-12599
```
The idea of an image bump is to use the exact image that was created by this PR

**Review by k15r — APPROVED** [review]

**Pull request merged**